src/mbed_tools/build/_internal/cmake_file.py

```diff
@@ -69,6 +71,7 @@ def _render_mbed_config_cmake_template(
         "macros": sorted(macros, key=lambda macro: macro.name),
         "max_name_length": max(_max_attribute_length(options, "macro_name"), _max_attribute_length(macros, "name")),
         "max_value_length": max(_max_attribute_length(options, "value"), _max_attribute_length(macros, "value")),
+        "baremetal_profile": "OFF"
```

Comment:
Could you explain how this can change to "ON"?

Reply:
This flag ON/OFF based on "requires": "bare-metal" mbed_app.json config as like previous tools even I asked clarification in a mail sometimes ago.

mbedos-tools is yet to add support for "requires" config.

Reply:
Resolving this as `mbed-tools` will no longer provide a CMake variable to indicate whether the baremetal profile was selected or not.